.github/workflows/ci-build.yml

```diff
@@ -146,142 +83,80 @@ jobs:
+          target: "all"
 
   docs-build:
     name: Building Documentation
     runs-on: ubuntu-latest
     needs: [docs-style]
 
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Setup Python
-        uses: actions/setup-python@v4
+      - name: Build documentation
+        uses: pyansys/actions/doc-build@v1
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          dependencies: "build-essential zip pandoc texlive-latex-extra latexmk texlive-pstricks"
 
-      - name: Build Documentation
-        run: |
-          pip install .[docs]
-          make -C doc html
-          touch doc/build/html/.nojekyll
-
-      - name: Upload Documentation
-        uses: actions/upload-artifact@v3
-        with:
-          name: Documentation
-          path: doc/build/html
-          retention-days: 7
-
-      - name: Deploy Docs
-        uses: JamesIves/github-pages-deploy-action@v4.4.1
+      - name: Deploy the latest documentation
+        uses: pyansys/actions/doc-deploy-dev@v1
         if: startsWith(github.event.ref, 'refs/tags')
         with:
-          TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: doc/build/html
-          CLEAN: true
+          cname: ${{ env.DOCUMENTATION_CNAME }}
+          token: ${{ secrets.GITHUB_TOKEN }}
 
   package:
     name: Package library
     runs-on: ubuntu-latest
-    needs: [docs-build, core-import]
+    needs: [docs-build, smoke-tests-core]
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: Build library source and wheel artifacts
+        uses: pyansys/actions/build-library@v1
         with:
+          library-name: ${{ env.PACKAGE_NAME }}
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
 
-      - name: Cache pip
-        uses: actions/cache@v3
-        with:
-          path: ~/.cache/pip
-          key: Python-${{ runner.os }}-${{ env.MAIN_PYTHON_VERSION }}-${{ hashFiles('pyproject.toml') }}-core
-          restore-keys: |
-            Python-${{ runner.os }}-${{ env.MAIN_PYTHON_VERSION }}
-
-      - name: Install dependencies and build the library
-        run: |
-          python -m pip install --upgrade pip poetry twine
-          python -m poetry build
-          python -m twine check dist/*
-
-      - name: Install PyAnsys from wheel
-        run: |
-          pip install dist/*.whl
-
-      - name: Upload wheel and binaries
-        uses: actions/upload-artifact@v3
-        with:
-          name: PyAnsys-packages
-          path: dist/
-          retention-days: 7
-
   release:
     name: Release project
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
-    needs: [package, extras-import]
+    needs: [package, smoke-tests-extras, wheelhouse-all]
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: Release to the private PyPI repository
```

Comment:
```suggestion
      - name: Release to PyPI repository
```

Reply:
@RobPasMue . I think we want to keep publishing to public PyPI but I must have forgotten a point.

Reply:
If not, you might have to change the action used below.

Reply:
Damn, you're totally right. I must have missed it.